### ISSUE TITLE
Leader schedule is not re-generated when a staked node changes their vote account node_pubkkey

### ISSUE THREAD

**Issue by mvines** [solana-labs/solana#12869]
STR:
1. Start a staked validator
2. Use `solana vote-update-validator` to update the node pubkey for the validator's vote account
3. Restart the staked validator with it's new identity keypair

The staked validator will continue to vote normally but will not produce any blocks as leader (presumably until the next leader schedule is generated).  

Should we apply the same update restrictions for the node pubkey as the authorized voter pubkey?

**Comment by mvines** [comment]
@carllin - wdyt we should do here?

**Comment by carllin** [comment]
@mvines yeah I think using the same restrictions for the node pubkey as the authorized voter pubkey makes sense, it's logically tidy without the need for any additional state, and straightforward to reason about.

**Comment by mvines** [comment]
Woot, so we get to rev the vote account state and can sneak in https://github.com/solana-labs/solana/issues/11799 at the same time

**Comment by mvines** [comment]
@ryoqun - are you interested in taking this on?

**Comment by mvines** [comment]
Yes, it's still a limitation
